src/com/github/igotyou/FactoryMod/interactionManager/FurnCraftChestInteractionManager.java

```diff
@@ -21,21 +29,137 @@
 public class FurnCraftChestInteractionManager implements IInteractionManager {
 	private FurnCraftChestFactory fccf;
 	private HashMap<Clickable, InputRecipe> recipes = new HashMap<Clickable, InputRecipe>();
+	private ReinforcementManager rm;
 
 	public FurnCraftChestInteractionManager(FurnCraftChestFactory fccf) {
 		this.fccf = fccf;
+		prepCitadel();
 	}
 
 	public FurnCraftChestInteractionManager() {
+		prepCitadel();
 	}
 
 	public void setFactory(FurnCraftChestFactory fccf) {
 		this.fccf = fccf;
 	}
+	
+	private void prepCitadel() {
+		if (FactoryModPlugin.getManager().isCitadelEnabled()) {
+			rm = Citadel.getReinforcementManager();
+		} else {
+			rm = null;
+		}
+	}
 
 	public void redStoneEvent(BlockRedstoneEvent e) {
-		// TODO
+		if (rm != null) {
+			// Note this also accomplishes all the Citadel checking we need.
+			BlockFace powerFace = findPoweringFace(e.getBlock(), fccf.getMultiBlockStructure().getAllBlocks());
+			if (powerFace != null) {
+				int trueNewCurrent = e.getBlock().getBlockPower(powerFace);
+				if (trueNewCurrent != e.getNewCurrent()) {
+					e.setNewCurrent(trueNewCurrent);
```

Comment:
My thoughts exactly. I'm not too worried either way, but if we can set it, all the better.